pkg/ccl/backupccl/restore_job.go

```diff
@@ -604,6 +604,11 @@ func restore(
 		return emptyRowCount, nil
 	}
 
+	details := job.Details().(jobspb.RestoreDetails)
+	if alreadyMigrated := checkForMigratedData(details, dataToRestore); alreadyMigrated {
```

Comment:
I _think_ I now understand why this early return if _any_ table has been migrated is safe -- we call `restore` then we migrate the restored system tables later, outside `restore`, right?

maybe a comment? or maybe we should just skip calling `restore` at all on resume if dataToRestore is empty or migratedData is true instead of early returning? or maybe I'm just being dense today, that could be true too.

in any case, totally non-blocking, just rambling out-loud here.

Reply:
The only catch here is that we should check for migrations per-batch. If the earlier batch (that restores the zones) needs a migration, that migration should happen before the restoration of the main data. We'll probably need something like this when re-keying the contents of system keys.

I kept it inside the restore since it was common to both the attempts of the "preData" restoration (that of the zones table), and the main data restoration.